### PR TITLE
Enable Temporal to startup even if Elastic Search is down

### PR DESCRIPTION
**Pull request by mastermanu**
A customer discovered an issue where Temporal Server will fail to start up if it is using ElasticSearch, and ElasticSearch is down. There are two separate issues:

1) Within the Server itself, we block Temporal from starting up if we cannot create an Elastic Search Client
2) Within the startup of the Docker Image, we wait indefinitely until we are able to access ElasticSearch.

This PR provides a proper fix for the first issue and a workaround for the second issue.

For the first issue, when we create the ElasticSearch client, we disable health checks, and then re-enable the healthchecks after the client is created. This ensures we can create the client even if ES is down.

For the second issue, we added another environment variable called "$BEST_EFFORT_CREATE_ES_SCHEMA." By default, this is false (meaning if ES is down, startup will block). If the user finds their startup is blocked because of this, they can set the variable to true. With a value of true, the docker script will continue after 30 seconds if it is unable to establish a connection to Elastic Search (we could also parameterize the 30 seconds as well).

We have a task at Temporal tracking the creation of a better story for the second issue.

Tested both fixes using local development environment and auto-setup containers. Also validated that ES catches up once it is started after Temporal has already started.

### CODE REVIEW

File: docker/start.sh [hunk 1]
```diff
@@ -119,16 +120,29 @@ wait_for_postgres() {
 }
 
 
-wait_for_es() {
+setup_es() {
+    SECONDS=0
+    duration=0
```
Comment (markmark206) [inline docker/start.sh:125]: we probably don't need this line, but it doesn't hurt either.

**Review by markmark206 — APPROVED** [review]

File: common/elasticsearch/client.go [hunk 2]
```diff
@@ -90,12 +90,23 @@ func NewClient(config *Config) (Client, error) {
...
 		elastic.SetDecoder(&elastic.NumberDecoder{}), // critical to ensure decode of int64 won't lose precise
 	)
+
 	if err != nil {
 		return nil, err
 	}
+
+	// Re-enable the healthcheck after client has successfully been created.
+	client.Stop()
+	elastic.SetHealthcheck(true)(client)
```
Comment (alexshtin) [inline common/elasticsearch/client.go:107]: What a syntax!

**Review by alexshtin — APPROVED** [review]

**Comment by wxing1292** [comment]
FYI, creating yet another env (`BEST_EFFORT_CREATE_ES_SCHEMA`) to support may not be a good idea, plus if ES cannot start during docker all in one setup, we should stop IMO.

**Comment by wxing1292** [comment]
> FYI, creating yet another env (`BEST_EFFORT_CREATE_ES_SCHEMA`) to support may not be a good idea, plus if ES cannot start during docker all in one setup, we should stop IMO.

turns out that the docker "check ES being online" is not only used by all in one setup for dev.
plz consider rename the `BEST_EFFORT_CREATE_ES_SCHEMA` env var

**Review by wxing1292 — APPROVED** [review]

File: docker/start.sh (hunk 1 above)
Comment (mastermanu) [reply docker/start.sh]: resolved

**Pull request merged**